src/modules/launcher/Plugins/Microsoft.Plugin.Settings/Main.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Plugin.Settings.Enumerations;
+using Microsoft.Plugin.Settings.Helper;
+using Microsoft.Win32;
+using Wox.Plugin;
+
+namespace Microsoft.Plugin.Settings
+{
+    public sealed class Main : IPlugin
+    {
+        private IEnumerable<IWindowsSetting>? _settingsList;
+
+        public void Init(PluginInitContext context)
+        {
+            var currentWindowsVersion = ushort.MinValue;
+
+            try
+            {
+                var releaseIdString = Registry.GetValue(
+                    "HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\Windows NT\\CurrentVersion",
+                    "ReleaseId",
+                    null) as string;
+
+                if (!ushort.TryParse(releaseIdString, out currentWindowsVersion))
+                {
+                    currentWindowsVersion = ushort.MinValue;
+                }
+            }
+            catch (Exception exception)
+            {
+                Debug.WriteLine(exception);
+
+                // fall-back
+                currentWindowsVersion = ushort.MinValue;
+            }
+
+            _settingsList = SettingsHelper.GetSettings();
+
+            // remove deprecated settings Windows settings
+            _settingsList = _settingsList
+                .Where(found => currentWindowsVersion < found.DeprecatedInVersion);
+
+            // remove settings that are for a higher Windows versions
+            _settingsList = _settingsList
+                .Where(found => currentWindowsVersion >= found.IntroducedInVersion);
```

Comment:
join these two where into the one, no need to iterate this list twice
`.Where(found => currentWindowsVersion < found.DeprecatedInVersion && currentWindowsVersion >= found.IntroducedInVersion);`